datajoint/user_relations.py

```diff
@@ -2,59 +2,83 @@
 Hosts the table tiers, user relations should be derived from.
 """
 
-import abc
 from .base_relation import BaseRelation
 from .autopopulate import AutoPopulate
 from .utils import from_camel_case
 from . import DataJointError
 
 
-class Part(BaseRelation, metaclass=abc.ABCMeta):
+class classproperty:
```

Comment:
What I had not realized in my ignorance was that to downstream users, class properties can be used the same way as instance properties.  At the time, I was under the impression that class properties always had to be called on the class and could not be called on the instance.  I also did not realize that when subclassing, each subclass gets its own class property and not just one shared across all subclasses.  These features, combined, make it acceptable to use classproperties.  Sorry I did not know this.  I thought classproperty would introduce all sorts of complications but after checking everything, it seems safe.